File: scratch/fix_website.py
import os
import glob

replacements = {
    "LUMI AI Engine": "AAIA",
    "LUMI AI Platform": "AAIA",
    "LUMI AI Solutions": "AAIA",
    "LUMI AI": "AAIA",
    "2.5 million": "20M+",
    "Godwin Achim Tetteh": "Achim Godwin Tetteh",
}

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        original_content = content
        
        for old, new in replacements.items():
            content = content.replace(old, new)
            
        if content != original_content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Updated {filepath}")
    except Exception as e:
        pass # Ignore unreadable files like binaries

base_path = r"c:\Users\hp\.gemini\antigravity\scratch\website-achtrex"
for root, dirs, files in os.walk(base_path):
    if "node_modules" in root or ".git" in root or ".next" in root:
        continue
    for file in files:
        if file.endswith((".tsx", ".ts", ".js", ".md", ".json")):
            process_file(os.path.join(root, file))

print("Replacements completed.")

File: scratch/fix_website.js
const fs = require('fs');
const path = require('path');

const replacements = {
    "AAIA": "AAIA",
    "AAIA": "AAIA",
    "AAIA": "AAIA",
    "AAIA": "AAIA",
    "20M+": "20M+",
    "Achim Godwin Tetteh": "Achim Godwin Tetteh",
};

function processFile(filepath) {
    try {
        let content = fs.readFileSync(filepath, 'utf8');
        let originalContent = content;
        
        for (const [old, newStr] of Object.entries(replacements)) {
            // using split and join to replace all occurrences
            content = content.split(old).join(newStr);
        }
        
        if (content !== originalContent) {
            fs.writeFileSync(filepath, content, 'utf8');
            console.log(`Updated ${filepath}`);
        }
    } catch (e) {
        // Ignore errors for unreadable or binary files
    }
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filepath = path.join(dir, file);
        if (file === 'node_modules' || file === '.git' || file === '.next') {
            continue;
        }
        const stat = fs.statSync(filepath);
        if (stat.isDirectory()) {
            walkDir(filepath);
        } else if (stat.isFile() && /\.(tsx|ts|js|md|json)$/.test(file)) {
            processFile(filepath);
        }
    }
}

const basePath = "c:\\Users\\hp\\.gemini\\antigravity\\scratch\\website-achtrex";
walkDir(basePath);
console.log("Replacements completed.");
